pre_commit/languages/docker.py

```diff
@@ -76,7 +78,28 @@ def install_environment(
         os.mkdir(directory)
 
 
+def _docker_is_rootless() -> bool:
+    returncode, stdout, stderr = cmd_output(
+        'docker', 'system', 'info',
+    )
+    for line in stdout.splitlines():
+        # rootless docker has "rootless"
+        # rootless podman has "rootless: true"
+        if line.strip().startswith('rootless'):
+            if 'false' not in line:
+                return True
+            break
+    return False
+
+
+@functools.lru_cache(maxsize=1)
+def docker_is_rootless() -> bool:
+    return _docker_is_rootless()
```

Comment:
Note: We had to do this to be able to test this with multiple mocked outputs. The cached version made that very hard.

Reply:
could also be something like `docker_is_rootless = functools.lru_cache(maxsize=1)(_docker_is_rootless)`

Reply:
Could. Not is if more or less readable.

Reply:
there's an example of testing the inner part of an `lru_cache` function in pre_commit/languages/node.py (tests/languages/node_test.py) -- basically, access the `__wrapped__` attribute of the cached object in the tests, then you don't need this indirection

Reply:
@themr0c @hroncok this will need to be addressed ^